packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -217,6 +217,47 @@ export const setSpanWithError = (
   span.setStatus(status);
 };
 
+/**
+ * Adds attributes for content-length and content-encoding HTTP headers
+ * @param { OutgoingHttpHeaders | IncomingHttpHeaders } headers http headers
+ * @param { Attributes } Attributes span attributes
+ * @param { boolean } isRequest set true for setting request content-header
+ */
+export const setContentLengthAttributes = (
+  headers: OutgoingHttpHeaders | IncomingHttpHeaders,
+  attributes: Attributes,
+  isRequest: boolean
+) => {
+  let isCompressed = false;
+
+  if (headers['content-length'] === undefined) return;
+
+  if (
+    headers['content-encoding'] &&
+    headers['content-encoding'] !== 'identity'
+  ) {
+    isCompressed = true;
+  }
+
+  if (isCompressed) {
+    if (isRequest) {
+      attributes[HttpAttribute.HTTP_REQUEST_CONTENT_LENGTH] =
+        headers['content-length'];
```

Comment:
content length header is a string. This attribute should be a number. please convert

Reply:
Pushed a commit to verify attribute is a number.